README.md

```diff
@@ -195,6 +198,70 @@ We have a small collection of algorithms, one for every day of the month. Scroll
   ```
 - **Resources**
   -[Matrix: Determinant](https://www.mathsisfun.com/algebra/matrix-determinant.html)
+
+### December 11 - Printing matrix in spiral form
+- **Problem**
+  - Given a Matrix of size `M * N`, output its elements in spiral form.
+  - To better understand the problem statement, consider the following matrix
+  ```
+  01     02     03
+  04     05     06
+  07     08     09
+   ```
+  - The Spiral form of a matrix is as below.
+    ```
+    01 --> 02 --> 03 --> 06 --> 09 --> 08 --> 07 --> 04 --> 05
+    ```
+  - Move Right, Move Down, Move Left, Move Up till you cover all elements of matrix.
+- **Uses**
+  - Matrices are used much more in daily life than people would have thought. In fact it is in front of us every day when going to work, at the university and even at home.
+  - [Real Applications of Matrices](https://www.ukessays.com/essays/mathematics/application-of-matrices-in-real-life-problems.php)
+- **Resources**
+  - Matrices are represented using two dimensional arrays in any . So to access a single element in the matrix two “for loops”
+  - [Operations on Matrices](https://www.geeksforgeeks.org/different-operation-matrices/)
+
+
+### December 12 - Reversing a singly linked list
+- **Problem**
+  - Taking a singly linked as the input the output has to be another singly linked list having the reverse of the initial linked list
```

Comment:
Can be shortened to "Given a singly linked list, output a linked list containing the same elements but in reverse order. "
Also, 
Optional: Complete the task without creating a new list. I.E., reverse the linked list inplace.